src/Optimizers/Hyper.jl

```diff
@@ -0,0 +1,109 @@
+using SparseArrays
+using KaHyPar
+
+@kwdef struct HyperGraphPartitioning <: Optimizer
+    metric::Function = removedsize
+    outer::Bool = false
+end
+
+
+function get_hypergraph(path)
+    # assume unique indices across all tensors
+    all_indices = mapreduce(head, ∪, path)
```

Comment:
Sorry, I had a typo in my suggestion.
```suggestion
    all_indices = mapreduce(head, ∪, path.args)
```